models/models_methods.go

```diff
@@ -357,6 +363,12 @@ func (with *WithModel) Validate(workflowID string, containers, services map[stri
 			return warnings, err
 		}
 
+		if stepID == StepListItemWithKey {
+			return warnings, errors.New("'with' group is not allowed in a 'with' group's step list")
+		} else if strings.HasPrefix(stepID, StepListItemStepBundleKeyPrefix) {
+			return warnings, errors.New("step bundle is not allowed in a 'with' group's step list")
```

Comment:
The workflow id is available as parameter here so we could add it to the error message. This still does not tell the user exactly which with group has an issue but at least it narrows it down to the workflow which owns the with group.